### DISCUSSION TITLE
[Feature request] jrpc set hostname and path instead of the full URL

### DISCUSSION THREAD

**Discussion by wiardvanrij** [fortio/fortio#639] [General]
At the moment I'm doing something like this:

```
func getJrpcDest(fooToken string) *jrpc.Destination {
	dest := &jrpc.Destination{
		Headers: &http.Header{
			"Authorization": {"Bearer " + fooToken},
		},
	}

	return dest
}
```

Which is useful as that sets the object which I then can re-use by setting just the URL. For example:

`jrpcDest = getJrpcDest(fooToken)`

followed by 

```
jrpcDest.URL = http://foo.bar/some/path
foo, err := jrpc.GetArray[Foo](jrpcDest)
```

It would be nice to allow to split URL into a host and a path function. This way I can improve my initial 'object' by including the host. Then on every usage, I can merely include the path instead of the full URL. This improves usability and allows you to set a single jrcp object that contains the needed header(s) and hostname.

**Comment by ldemailly** [comment]
This sounds a tiny bit more like a https://github.com/fortio/fortio/discussions than an issue but let's keep it here for now/now that it is here

let's say you're making a client library for a server X that has 
```
/foo
/bar
/blah
```
endpoints requiring a token


I think you would do something like:

```golang
type XCli struct {
	BaseServer string
	Token      string
}

type Foo struct{} // replace with actual result from foo call

type Bar struct{} // replace with actual result from bar call

func (x *XCli) makeDest(path string) *jrpc.Destination {
	return &jrpc.Destination{
		Headers: &http.Header{
			"Authorization": {"Bearer " + x.Token},
		},
		URL: x.BaseServer + path,
	}
}

func (x *XCli) Foo() (*Foo, error) {
	return jrpc.Get[Foo](x.makeDest("/foo"))
}

func (x *XCli) Bar() (*Bar, error) {
	return jrpc.Get[Bar](x.makeDest("/bar"))
}
//...
```

https://go.dev/play/p/PEQDJ5dl172

**Comment by ldemailly** [comment]
going to close this for now as I think `+` isn't worth the complexity of adding more to the api
